saleor/webhook/payloads.py

```diff
@@ -1473,3 +1485,44 @@ def generate_transaction_action_request_payload(
         "meta": generate_meta(requestor_data=generate_requestor(requestor)),
     }
     return json.dumps(payload, cls=CustomJsonEncoder)
+
+
+@traced_payload_generator
+def generate_thumbnail_payload(
+    thumbnail: Thumbnail,
+    instance: Union["User", "Category", "Collection", "ProductMedia"],
+):
+    thumbnail_url = prepare_thumbnail_file_name(thumbnail.image.name, thumbnail.size, thumbnail.format)
+    type = instance.__class__.__name__
+
+    serializer = PayloadSerializer()
+    object_payload = {
+        "type": type,
+        "id": graphene.Node.to_global_id(type, instance.id)
+    }
+    payload = serializer.serialize(
+        [thumbnail],
+        fields=("size", "format"),
+        extra_dict_data={
+            "url": thumbnail_url,
+            "url_origin": thumbnail.image.url,
```

Comment:
I'm not sure we should introduce "static" payloads for new events, we should rather support only subscriptions. These could be as simple as possible, for example only ID of the changed object, so that the listener app could fetch more details for API. I would not include any DB relations here for optimal DB performance.